File: Go/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.go
func solution(array []int) int {
    // make map
    counts := make(map[int]int)
    
    // counting
    for _, v := range array {
        counts[v]++
    }
    
    // init values
    answer := -1
    maxValue := 0
    
    // search mode
    for n, v := range counts {
        if maxValue == v {
            answer = -1
        }
        if v > maxValue {
            maxValue = v
            answer = n
        }
    }
    
    return answer
}

File: Java/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] array) {
        Map<Integer, Long> counts = Arrays.stream(array)
            .boxed()
            .collect(
            Collectors.groupingBy(x -> x, Collectors.counting())
        );
        
        int answer = 0;
        long maxCount = 0;
        
        for(Map.Entry<Integer, Long> c: counts.entrySet()) {
            if (c.getValue() == maxCount) answer = -1;
            if (c.getValue() > maxCount) {
                answer = c.getKey();
                maxCount = c.getValue();
            }
        }
        
        return answer;
    }
}